File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/operations/getamendmentsbysubscriptionid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetAmendmentsBySubscriptionIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription-id" })
  subscriptionId: string;
}


export class GetAmendmentsBySubscriptionIdHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Entity-Ids" })
  zuoraEntityIds?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=Zuora-Track-Id" })
  zuoraTrackId?: string;
}


export class GetAmendmentsBySubscriptionIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetAmendmentsBySubscriptionIdPathParams;

  @SpeakeasyMetadata()
  headers: GetAmendmentsBySubscriptionIdHeaders;
}


export class GetAmendmentsBySubscriptionIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  getAmendmentType?: shared.GetAmendmentType;

  @SpeakeasyMetadata()
  headers: Record<string, string[]>;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/operations/get_amendmentsbysubscriptionid.py
import dataclasses
from typing import Optional
from ..shared import getamendmenttype as shared_getamendmenttype


@dataclasses.dataclass
class GetAmendmentsBySubscriptionIDPathParams:
    subscription_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'subscription-id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetAmendmentsBySubscriptionIDHeaders:
    zuora_entity_ids: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Entity-Ids', 'style': 'simple', 'explode': False }})
    zuora_track_id: Optional[str] = dataclasses.field(default=None, metadata={'header': { 'field_name': 'Zuora-Track-Id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetAmendmentsBySubscriptionIDRequest:
    headers: GetAmendmentsBySubscriptionIDHeaders = dataclasses.field()
    path_params: GetAmendmentsBySubscriptionIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class GetAmendmentsBySubscriptionIDResponse:
    content_type: str = dataclasses.field()
    headers: dict[str, list[str]] = dataclasses.field()
    status_code: int = dataclasses.field()
    get_amendment_type: Optional[shared_getamendmenttype.GetAmendmentType] = dataclasses.field(default=None)
    
